Dockerfile

```diff
@@ -44,3 +48,21 @@ RUN printf '#!/bin/sh\nipfs --api /ip4/`getent ahostsv4 host.docker.internal | g
 RUN chmod u+x entrypoint.sh
 
 ENTRYPOINT [ "./entrypoint.sh" ]
+
+# -------------------------------------
+# Build Option 2: Host the app remotely
+# -------------------------------------
+
+FROM base as remote-host
+RUN apt-get update && apt-get install -y curl
+
+# add the build output to IPFS, write the hash to a file and print during build
+RUN ipfs add --cid-version 1 --quieter --recursive /export > ipfs_hash.txt && cat ipfs_hash.txt
+
+# nft.storage api key required, pass the key during runtime with -e flag
+ENV NFTSTORAGE_API_KEY=
+
+# generate the car file and upload to nft.storage through http endpoint
+RUN printf '#!/bin/sh\nif [ -z "$NFTSTORAGE_API_KEY" ] || [ "$NFTSTORAGE_API_KEY" = "" ]; then echo "Error: NFTSTORAGE_API_KEY is not set or is blank"; exit  1; fi\nipfs dag export $(cat ipfs_hash.txt) > output.car\ncurl -X POST -H "Authorization: Bearer $NFTSTORAGE_API_KEY" -H "Content-Type: application/car" --data-binary @output.car https://api.nft.storage/upload' >> entrypoint.sh && chmod u+x entrypoint.sh
+
+ENTRYPOINT [ "./entrypoint.sh" ]
```

Comment:
We should consider moving this into separate RUN lines and not having an `ENTRYPOINT`.  This would cause the docker image build process to submit to nft.storage.  We would then discard the image after that.  Essentially it just makes it so the image build process is effectively executing some code inside a container.  The nice thing about this strategy is that it keeps all of the build scripts inside a `Dockerimage` file and easy to read rather than spread out.

Reply:
Another option (we can discuss merits on Discord) would be to have a single image and a more complex script that lets you do *either* local deploy or remote deploy.  We would publish the image and then the last CI step would be to run the built image with the parameter to trigger remote deploy.

This second option is growing on me.